src/dmd/expressionsem.d

```diff
@@ -3617,7 +3617,12 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 auto dse = new DsymbolExp(loc, cd.vtblSymbol());
                 auto ase = new AddrExp(loc, dse);
                 auto pte = new DotIdExp(loc, new ThisExp(loc), Id.__vptr);
-                auto ate = new AssignExp(loc, pte, ase);
+                Expression ate = new AssignExp(loc, pte, ase);
+
+                // skip the vptr assignment during CTFE; it's not supported and superfluous,
+                // as an external super in C++ is non-CTFE-able anyway
+                // => __ctfe || cast(void) (this.__vptr = __vtbl)
+                ate = new LogicalExp(loc, TOK.orOr, new IdentifierExp(loc, Id.ctfe), new CastExp(loc, ate, Type.tvoid));
```

Comment:
I feel like a potentially superior implementation would be to detect if `super()` is extern, and then skip the assignment on that basis. That way, even outside of CTFE, the redundant work to assign the vtable will be skipped in cases when it doesn't need to.